src/fsharp/service/ServiceUntypedParse.fs

```diff
@@ -932,18 +934,15 @@ module UntypedParseImpl =
         | ParsedInput.ImplFile input -> walkImplFileInput input
 
     type internal TS = AstTraversal.TraverseStep
+    /// Matches the most nested [< and >] pair.
+    let insideAttributeApplicationRegex = Regex(@"(?<=\[\<)(?<attribute>(.*?))(?=\>\])", RegexOptions.Compiled ||| RegexOptions.ExplicitCapture)
 
     /// Try to determine completion context for the given pair (row, columns)
-    let TryGetCompletionContext (pos, untypedParseOpt: FSharpParseFileResults option, lineStr: string) : CompletionContext option = 
-        let parsedInputOpt =
-            match untypedParseOpt with
-            | Some upi -> upi.ParseTree
-            | None -> None
+    let TryGetCompletionContext (pos, parsedInputOpt: ParsedInput option, lineStr: string) : CompletionContext option = 
```

Comment:
Can we go further and do `parsedInput : ParsedInput` instead of `parstedInputOpt : ParsedInput option` ?

Reply:
Yes.